types/lodash/common/lang.d.ts

```diff
@@ -1224,14 +1224,14 @@ declare module "../index" {
          * @param value The value to check.
          * @return Returns true if value is an object, else false.
          */
-        isObject(value?: any): boolean;
+        isObject(value?: any): value is object;
     }
 
     interface LoDashImplicitWrapper<TValue> {
         /**
          * see _.isObject
          */
-        isObject(): boolean;
+        isObject(): this is LoDashImplicitWrapper<object>;
```

Comment:
This doesn't seem to do much based on your tests... I would be fine sticking with `boolean`. Although I suppose there are some uncommon cases where this could be useful, and it's not hurting anything, so you can keep it if you want.